src/RuleSetInterface.php

```diff
@@ -48,6 +48,15 @@ public function getRules();
      */
     public function getSetDefinitionNames();
 
+    /**
+     * Get the related description of the set name.
+     *
+     * @param string $setDefinitionName
+     *
+     * @return string
+     */
+    public function getSetDescription($setDefinitionName);
```

Comment:
This is public interface, this is BC breaker.

Also, I don't believe that having it here is the best approach. 

I'm aware that having it fully decoupled is a bigger thing to do (like one ruleset definition = one class), are you willing to try do this? If not, let us still decouple descriptions into autonomic, dedicated, internal class

Reply:
> one ruleset definition = one clas

This might help implementing custom rulesets. Worth the try IMO :+1: